greenpeace/planet4-master-theme
# PLANET-7578: Migrate Content Covers Blocks to Posts List blocks

Ref: https://jira.greenpeace.org/browse/PLANET-7578

<hr>

**DESCRIPTION:**
This PR migrates the Content Covers blocks to Posts List blocks.

<hr>

**TESTING INSTRUCTIONS:**
1. Run this PR in your local environment.
2. Add a Content Cover block to a post or page. it won't work on the backend because of this PR's changes but it's good enough to test the migration script. You can also directly copy/paste this in the code editor:

```

<div class="wp-block-planet4-blocks-covers"><div data-hydrate="planet4-blocks/covers" data-attributes="{&quot;cover_type&quot;:&quot;content&quot;,&quot;initialRowsLimit&quot;:1,&quot;title&quot;:&quot;Content Cover #0&lt;br&gt;This is the title&quot;,&quot;description&quot;:&quot;This is the content cover description.&lt;br&gt;Layout: grid&lt;br&gt;Rows: 1&lt;br&gt;Tags: -&lt;br&gt;Post types: -&lt;br&gt;Manual override: 3 pages&quot;,&quot;version&quot;:2,&quot;tags&quot;:[],&quot;post_types&quot;:[],&quot;posts&quot;:[256,810,1113],&quot;layout&quot;:&quot;grid&quot;,&quot;isExample&quot;:false,&quot;readMoreText&quot;:&quot;More and more&quot;,&quot;covers&quot;:[]}"></div></div>



<div class="wp-block-planet4-blocks-covers"><div data-hydrate="planet4-blocks/covers" data-attributes="{&quot;cover_type&quot;:&quot;content&quot;,&quot;initialRowsLimit&quot;:1,&quot;title&quot;:&quot;Content Cover #1&lt;br&gt;This is the title&quot;,&quot;description&quot;:&quot;This is the content cover description.&lt;br&gt;Layout: grid&lt;br&gt;Rows: 1&lt;br&gt;Tags: renewables, consumption&lt;br&gt;Post types: press release&quot;,&quot;version&quot;:2,&quot;tags&quot;:[7,8],&quot;post_types&quot;:[14],&quot;posts&quot;:[],&quot;layout&quot;:&quot;grid&quot;,&quot;isExample&quot;:false,&quot;readMoreText&quot;:&quot;More and more&quot;,&quot;covers&quot;:[]}"></div></div>



<div class="wp-block-planet4-blocks-covers"><div data-hydrate="planet4-blocks/covers" data-attributes="{&quot;cover_type&quot;:&quot;content&quot;,&quot;initialRowsLimit&quot;:2,&quot;title&quot;:&quot;Content Cover #2&lt;br&gt;This is the title&quot;,&quot;description&quot;:&quot;This is the content cover description.&lt;br&gt;Layout: grid&lt;br&gt;Rows: 2&lt;br&gt;Tags: ocean, climate, consumption, renewables&lt;br&gt;Post types: -&quot;,&quot;version&quot;:2,&quot;tags&quot;:[26,6,8,7],&quot;post_types&quot;:[],&quot;posts&quot;:[],&quot;layout&quot;:&quot;grid&quot;,&quot;isExample&quot;:false,&quot;readMoreText&quot;:&quot;&quot;,&quot;covers&quot;:[]}"></div></div>



<div class="wp-block-planet4-blocks-covers"><div data-hydrate="planet4-blocks/covers" data-attributes="{&quot;cover_type&quot;:&quot;content&quot;,&quot;initialRowsLimit&quot;:1,&quot;title&quot;:&quot;Content Cover #3&lt;br&gt;This is the title&quot;,&quot;description&quot;:&quot;This is the content cover description.&lt;br&gt;Layout: carousel&lt;br&gt;Tags: ocean, climate, consumption, renewables&lt;br&gt;Post types: -&quot;,&quot;version&quot;:2,&quot;tags&quot;:[26,6,8,7],&quot;post_types&quot;:[],&quot;posts&quot;:[],&quot;layout&quot;:&quot;carousel&quot;,&quot;isExample&quot;:false,&quot;readMoreText&quot;:&quot;&quot;,&quot;covers&quot;:[]}"></div></div>



<div class="wp-block-planet4-blocks-covers"><div data-hydrate="planet4-blocks/covers" data-attributes="{&quot;cover_type&quot;:&quot;content&quot;,&quot;initialRowsLimit&quot;:1,&quot;title&quot;:&quot;Content Cover #4&lt;br&gt;This is the title&quot;,&quot;description&quot;:&quot;This is the content cover description.&lt;br&gt;Layout: carousel&lt;br&gt;Tags: &lt;br&gt;Post types: story&quot;,&quot;version&quot;:2,&quot;tags&quot;:[],&quot;post_types&quot;:[16],&quot;posts&quot;:[],&quot;layout&quot;:&quot;carousel&quot;,&quot;isExample&quot;:false,&quot;readMoreText&quot;:&quot;&quot;,&quot;covers&quot;:[]}"></div></div>

```

4. Run the migrator command: `npx wp-env run cli wp p4-run-activator`
5. Check whether the migration worked as expected.
6. To reset the environment database back to the latest default content and repeat the tests, run: `npm run db:reset`.
